android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -539,6 +548,10 @@ open class FxaAccountManager(
             resetAccount()
             Event.Progress.LoggedOut
         }
+        ProgressState.DeletingAccount -> {
+            resetAccount()
```

Comment:
You may need to use/make a separate method. `resetAccount` calls `account.disconnect()` which wouldn't make sense to call if we are receiving the web channel response that the account is deleted.

Reply:
From the [doc](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/concept/sync/src/main/java/mozilla/components/concept/sync/OAuthAccount.kt#190):

     * Reset internal account state and destroy current device record.
     * Use this when device record is no longer relevant, e.g. while logging out. On success, other
     * devices will no longer see the current device in their device lists.
     
     Looks pretty relevant #3825

Reply:
When the account has been deleted, the device record is not longer relevent, so we should clear the device record.

Reply:
Correct, we do that with `accountStorage.clear()` locally. Why do we need to tell the server to delete the device record as well (after we received the account deletion)?

Reply:
ah, sorry, missed this ping - IIUC, the account deleted notification means the account *has been* deleted. I expect any attempts to do anything, including revoking the device record, would fail.

Reply:
(I haven't really looked at this patch - we *should* delete the device record on local disconnect.)